cmd/microcloud/main_init.go

```diff
@@ -952,12 +952,34 @@ func (c *initConfig) setupCluster(s *service.Handler) error {
 
 		if !shared.ValueInSlice(profile.Name, profiles) {
 			err = lxdClient.CreateProfile(profile)
+			if err != nil {
+				return err
+			}
 		} else {
-			err = lxdClient.UpdateProfile(profile.Name, profile.ProfilePut, "")
-		}
+			// Ensure any pre-existing devices and config are carried over to the new profile, unless we are managing them.
+			existingProfile, _, err := lxdClient.GetProfile("default")
+			if err != nil {
+				return err
+			}
 
-		if err != nil {
-			return err
+			for k, v := range profile.Config {
+				_, ok := existingProfile.Config[k]
+				if !ok {
```

Comment:
Exiting would be a poor choice because we will always see a collision here, since we set those keys when running `microcloud init` already. 

I forgot that we decided to ask a question, I think having a question that lets the user choose whether we should overwrite conflicting profile devices would be the best approach.